bff/templates/deployment-sidecar.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
```

Comment:
Given that we do not have an optional `if` here, this will be a breaking change: if the bff values file does NOT set the sidecar related properties, the helm chart apply will fail.

IMO we should do it in a non-breaking change way: make the side car optional.

Example: https://github.com/PilotDataPlatform/helm-charts/blob/main/portal/templates/ingress.yaml#L1